Common/include/option_structure.hpp

```diff
@@ -2317,6 +2323,33 @@ static const MapType<std::string, POD_KIND> POD_Map = {
   MakePair("INCREMENTAL_POD", POD_KIND::INCREMENTAL)
 };
 
+/*!
+ * \brief Type of operation for the linear system solver, changes the source of solver options.
+ */
+enum class LINEAR_SOLVER_MODE {
+  STANDARD,        /*!< \brief Operate in standard mode. */
+  MESH_DEFORM,     /*!< \brief Operate in mesh deformation mode. */
+  GRADIENT_MODE,   /*!< \brief Operate in gradient smoothing mode. */
+};
+
+/*!
+ * \brief mode of operation for the sobolev smoothing solver.
+ */
+enum ENUM_SOBOLEV_MODUS {
+ NO_MODUS = 0,                    /*!< \brief Default option if none is choosen. */
+ PARAM_LEVEL_COMPLETE   = 1,      /*!< \brief Operate on parameter level. */
+ MESH_LEVEL  = 2,                 /*!< \brief Operate on mesh level. */
+ DEBUG = 3,                       /*!< \brief Special flag for debugging. */
+ ONLY_GRAD = 4,                   /*!< \brief Flag for OneShot to only compute the original gradient. */
```

Comment:
enum class please, and without explicit numbers (for consistency with recent changes)